core/src/tangram.cpp

```diff
@@ -63,19 +62,35 @@ void initialize(const char* _scenePath) {
         logMsg("Loading Tangram scene file: %s\n", sceneRelPath.c_str());
         auto sceneString = stringFromResource(sceneRelPath.c_str());
 
-        SceneLoader loader;
+        if (SceneLoader::loadScene(sceneString, *m_scene)) {
+            m_tileManager->setScene(m_scene);
 
-        loader.loadScene(sceneString, *m_scene, *m_tileManager, *m_view);
+            glm::dvec2 projPos = m_view->getMapProjection().LonLatToMeters(m_scene->startPosition);
+            m_view->setPosition(projPos.x, projPos.y);
+            m_view->setZoom(m_scene->startZoom);
+        }
 
         m_skybox = std::unique_ptr<Skybox>(new Skybox("img/cubemap.png"));
         m_skybox->init();
-
     }
 
     logMsg("finish initialize\n");
 
 }
 
+void loadScene(const char* _scenePath) {
+    logMsg("Loading scene file: %s\n", _scenePath);
```

Comment:
A bit harder is to remove individual fonts from FontContext since fontstash has no method to remove fonts. We could move the FontContext to be a member of Scene though, but this could better go into the add-merged-font-style branch.